core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -67,6 +67,12 @@
   public static final String VALIDATE_CARBON_INPUT_SEGMENTS = "validate.carbon.input.segments.";
 
   /**
+   * Whether load/insert command is fired internally or by the user.
+   * Used to block load/insert on pre-aggregate if fired by user
+   */
+  public static final String IS_INTERNAL_LOAD_CALL = "is.internal.load.call";
```

Comment:
Seems no testcase for this option. And the option name should start with `carbon`

Reply:
This option/property will not be exposed to the user. It will be set by the post load listener to know whether the load is fired by the user or is it an internal call.

Test case is added in TestPreAggregateLoad